contracts/utils/test/ownable.ts

```
@@ -0,0 +1,65 @@
+import { BlockchainLifecycle } from '@0x/dev-utils';
+import { OwnableRevertErrors } from '@0x/utils';
+import * as chai from 'chai';
+import * as _ from 'lodash';
+
+import { artifacts, TestOwnableContract } from '../src';
+
+chaiSetup.configure();
+const expect = chai.expect;
+const blockchainLifecycle = new BlockchainLifecycle(web3Wrapper);
+
+describe('Ownable', () => {
+    let ownable: TestOwnableContract;
+    let owner: string;
+    let nonOwner: string;
+
+    before(async () => {
+        const accounts = await web3Wrapper.getAvailableAddressesAsync();
+        owner = await accounts[0];
+        nonOwner = await accounts[1];
+        await blockchainLifecycle.startAsync();
+        // Deploy SafeMath from the owner address
+        txDefaults.from = owner;
+        ownable = await TestOwnableContract.deployFrom0xArtifactAsync(
+            artifacts.TestOwnable,
+            provider,
+            txDefaults,
+        );
+    });
+
+    after(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+
+    // tslint:disable:no-unused-expression
+    describe('onlyOwner', () => {
+        it('should throw if sender is not owner', async () => {
+            const expectedError = new OwnableRevertErrors.OnlyOwnerError(
+                nonOwner,
+                owner,
+            );
+            return expect(ownable.externalOnlyOwner.callAsync({ from: nonOwner })).to.revertWith(expectedError);
+        });
+
+        it('should succeed if sender is owner', async () => {
+            expect(ownable.externalOnlyOwner.callAsync({ from: owner })).to.be.fulfilled;
+        });
+    });
+
+    describe('transferOwnership', () => {
+        it('should not transfer ownership if the specified new owner is the zero address', async () => {
+            expect(ownable.transferOwnership.sendTransactionAsync(constants.NULL_ADDRESS, { from: owner })).to.be.fulfilled;
```

Hmm not sure how I feel about this failing silently. It's probably outside the scope of this PR, but I've created a task in the protocol backlog to change this.

---

Yeah, I actually had a similar thought. I'm working on a PR that will disallow 0 in MixinAuthorizable as well, and I can fix it there.

---

```suggestion
            await expect(ownable.transferOwnership.sendTransactionAsync(constants.NULL_ADDRESS, { from: owner })).to.be.fulfilled('');
```